engineapi/engineapi/routes/configs.py

```diff
@@ -0,0 +1,161 @@
+import logging
+from typing import Any, Dict, List, Set
+
+from bugout.data import BugoutResource
+from fastapi import (
+    BackgroundTasks,
+    Body,
+    Depends,
+    FastAPI,
+    HTTPException,
+    Query,
+    Request,
+)
+from pydantic import AnyHttpUrl
+
+from .. import actions, data
+from ..middleware import (
+    BroodAuthMiddleware,
+    BugoutCORSMiddleware,
+    EngineHTTPException,
+    check_default_origins,
+    fetch_and_set_cors_origins_cache,
+    parse_origins_from_resources,
+)
+from ..settings import (
+    BUGOUT_REQUEST_TIMEOUT_SECONDS,
+    BUGOUT_RESOURCE_TYPE_APPLICATION_CONFIG,
+    DOCS_TARGET_PATH,
+    MOONSTREAM_ADMIN_USER,
+    MOONSTREAM_APPLICATION_ID,
+)
+from ..settings import bugout_client as bc
+from ..version import VERSION
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+tags_metadata = [
+    {"name": "configs", "description": "Moonstream Engine API configurations"}
+]
+
+whitelist_paths: Dict[str, str] = {}
+whitelist_paths.update(
+    {
+        "/configs/docs": "GET",
+        "/configs/openapi.json": "GET",
+    }
+)
+
+app = FastAPI(
+    title=f"Moonstream Engine API configurations",
+    description="Moonstream Engine API configurations endpoints.",
+    version=VERSION,
+    openapi_tags=tags_metadata,
+    openapi_url="/openapi.json",
+    docs_url=None,
+    redoc_url=f"/{DOCS_TARGET_PATH}",
+)
+
+
+app.add_middleware(BroodAuthMiddleware, whitelist=whitelist_paths)
+
+app.add_middleware(
+    BugoutCORSMiddleware,
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+
+@app.get("/cors", response_model=data.CORSResponse)
+async def get_cors(
+    request: Request,
+):
+    try:
+        resources = bc.list_resources(
+            token=request.state.token,
+            params={
+                "application_id": MOONSTREAM_APPLICATION_ID,
+                "type": BUGOUT_RESOURCE_TYPE_APPLICATION_CONFIG,
+                "setting": "cors",
+            },
+            timeout=BUGOUT_REQUEST_TIMEOUT_SECONDS,
+        )
+        resource_origins_set: Set[str] = parse_origins_from_resources(
+            resources.resources
+        )
+    except Exception as err:
+        logger.error(repr(err))
+        raise EngineHTTPException(status_code=500)
+
+    return data.CORSResponse(cors=",".join(list(resource_origins_set)))
+
+
+@app.put("/cors", response_model=data.CORSResponse)
```

Comment:
Is this a public endpoint? Are we not checking user credentials before adding CORS?

Reply:
We are using `token=request.state.token` in this endpoint.